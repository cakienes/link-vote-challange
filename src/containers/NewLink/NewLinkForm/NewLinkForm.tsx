import React from 'react';
import { reduxForm } from 'redux-form';
import Button from '../../../components/Button/Button';
import Input from '../../../components/Input/Input';
import INewLinkFormProps from './interface/INewLinkFormProps';
import './NewLinkForm.scss';

export const NewLinkForm = (props: INewLinkFormProps) => {
    return (
        <form onSubmit={props.handleSubmit}>
            <div className="row">
                <div className="col-12">
                    <div className="formName">Add New Link</div>
                </div>
                <div className="col-12">
                    <Input name="name" placeholder="e.g. Alphabet" label="Link Name:" />
                </div>
                <div className="col-12">
                    <Input name="link" placeholder="e.g. http://abc.xyz" label="Link URL:" />
                </div>
            </div>
            <div className="row">
                <div className="col">
                    <div className="buttonHolder">
                        <Button label="ADD" type="submit" />
                    </div>
                </div>
            </div>
        </form>
    );
};

export default reduxForm({
    form: 'newLinkForm',
})(NewLinkForm);
