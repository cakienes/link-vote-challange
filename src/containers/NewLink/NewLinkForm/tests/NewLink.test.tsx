import { shallow } from 'enzyme';
import React from 'react';
import { NewLinkForm } from '../NewLinkForm';

describe('NewLinkForm.tsx', () => {
    it('renders without crashing', async () => {
        expect(shallow(<NewLinkForm />)).toMatchSnapshot();
    });
});
