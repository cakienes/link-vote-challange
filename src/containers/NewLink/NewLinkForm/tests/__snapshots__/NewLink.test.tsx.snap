// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewLinkForm.tsx renders without crashing 1`] = `ShallowWrapper {}`;
